src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -951,11 +963,29 @@
                  (user-node->plan-node (user-query->logical-obj subquery-name)
                                        (first subquery-expression)))))))
 
-(pls/defn-validated columns->fields :- [(s/either s/Keyword SqlCall)]
+(pls/defn-validated columns->fields :- [(s/either s/Keyword SqlCall SqlRaw)]
   "Convert a list of columns to their true SQL field names."
   [query-rec
    columns :- [s/Str]]
-  (map #(get-in query-rec [:projections % :field])
+  ; This case expression here could be eliminated if we just used a projections list
+  ; and had the InExpression use that to generate the sql, but as it is the zipper we
+  ; use to walk the plan won't see instances of hashes and uuids among fields that have
+  ; gone through this function
+  (map #(let [projection (get-in query-rec [:projections %])]
+          (case (:type projection)
+            :uuid
```

Comment:
Similar point here actually, ie. if we had the conversion baked into the query-rec etc.